### ISSUE TITLE
[BUG] In Game Mode Touchscreen clicks does not act as dualsense touchpad clicks

### ISSUE THREAD

**Issue by Vallefor** [streetpea/chiaki-ng#481]
**Describe the bug**
When I use chiaki-ng in Game Mode on Rog Ally (using Bazzite OS) - touchscreen clicks does not act as dualsense touchpad clicks. Swipes and touch moves works fine tho. Just clicks do not work.
Can record a video if it helps.

**To Reproduce**
1. Run chiaki-ng in game mode on Rog Ally using Bazzite (last stable, 41v)
2. Connects to your PS5 and run Astros Play room for controller demonstration
3. Touchscreen clicks does not registered

**Expected behavior**
Touchscreen clicks should registered as dualsense touchpad clicks.

**Desktop (please complete the following information):**
 - OS: BazziteOS 
 - Version Stable 41.20241118
 - Device Rog Ally Z1 Extreme (not X)
 - chiaki-ng Version 1.9.1

**Comment by streetpea** [comment]
@Vallefor are you touching the outer 5% edge of the touchscreen?

**Comment by Vallefor** [comment]
@streetpea Sorry, was unable to demonstrate bug at the weekends. Can you reopen it?
I record video for you.

Desktop mode:
https://photos.app.goo.gl/rUpZzUfFYtGTycJu6

Game mode:
https://photos.app.goo.gl/MehYQ2zKi2qBFAge6

**Comment by streetpea** [comment]
I'm closing for now as it seems like you just tried the touchpad in a situation where the game didn't support it (Astro's Playroom controller demo) and so it didn't work. If you also have a situation where it doesn't work with a game/action that actually implements touchpad support @Vallefor please let me know and I'll reopen it.

**Comment by chino** [comment]
I'm running into the same issue. Is there a specific profile I should be using? I tried the chiaki4deck and chiaki-ng profiles. I can't seem to find one that would allow the touch screen to act like the pad.  To note I'm using a few day old setup of bazzite + chiaki-ng on a newly setup Ally Rog (non x) just like this post.

**Comment by streetpea** [comment]
@chino are you touching the edge of the screen and using a game with touchpad clicks?

**Comment by chino** [comment]
Yes, I'm playing The Last of Us which requires it for opening up the inventory in game. I've tried with the controller profiles as well as "without" steam controller using just raw inputs. Everything else works as expected as long as I'm running in Steam big screen mode. When I run in desktop mode then the screen reacts but all the button mappings are messed up at that point. I've tried using the various HHD (hand held daemon) profiles like Dual Sense, Xbox, and so forth with the various different settings.  I'd love to get to a point where even games that require "swipe" actions work too.

**Comment by chino** [comment]
Also what is the latest steam game controller profile that I should use to unlock the most features so it's as close to the ps5 controller as it can be?  I tried various ones that were in the list but struggled.

**Comment by streetpea** [comment]
@chino if you are in gaming mode you need to turn touchscreen mode on in the controller profile.

**Comment by streetpea** [comment]
I haven’t made any profiles for rog ally so none will show up bc I don’t have a rog ally

**Comment by streetpea** [comment]
@chino map it based off of the controller profile for steam deck in the docs but change as necessary for rog ally. The touchscreen native thing is this: https://streetpea.github.io/chiaki-ng/setup/controlling/#default-chiaki-ng-layout-touchscreen-mapping

**Comment by chino** [comment]
Ok thanks. Does that setup on steamdeck end up supporting "swipe" actions as well or just clicks? Via the touchscreen that is? The Ally does not have touchpads like the deck does, only the screen.

**Comment by streetpea** [comment]
@chino swipes and clicks. Clicks are outer 5% edges of touchscreen

**Comment by chino** [comment]
Ya, the one thing that's a little confusing is that the Ally maps the edge of the screen to various swipe actions. Like swiping from bottom brings up keyboard, from the right brings up the quick menu, etc. So I'm not sure if I'd need to disable those.

Do you know a game that supports swipe for some way for me to simply test if the swipe works? I don't seem to see it activate when I go into the steam controller test screen.

**Comment by streetpea** [comment]
@chino astros playroom controller demo does

**Comment by chino** [comment]
I was able to get it to work with the link you provided. I think this would have been turnkey if I had originally been able to find your community layout. The link still says to search chiaki4deck instead of chiaki-ng layouts? Ideally I'd like to use your layout if I can find it since it seems there are many other tweaks still on the page to get 100%.

Curious if the dual sense speaker gets forwarded to the main speakers or how that works?

**Comment by chino** [comment]
Do you think you could explain why the `Action Set -> Always on -> Touchpad native support` ends up working? Is it just making the touchscreen effectively a native "passthrough" directly to Chiaki which then Chiaki is smart enough to map that to the dual sense touchpad?

**Comment by streetpea** [comment]
@chino its because the original controller layout is for steam deck controller not rog ally controller so you won’t find it for rog ally. Touchpad native support works because in regular mode there is no touchscreen events it’s translated to mouse clicks but in touchscreen mode it actually sends the touchscreen events

**Comment by chino** [comment]
Right, I get that, but that assumes something is actually intercepting
those touchscreen events and translating them to touchpad events?

Note, I do see a bunch of chiaki-ng layouts and I can see the chiaki4deck
ones as well if I rename the program properly. However I was just never
able to find a newer layout with the full configs.

Either way I'm happy to help here to create a set of profiles that you
could upload as official for folks?

I will also say that these days I picked up my device for $300 pre-owned so
it's become cheaper. I imagine folks might be willing to pool together
funds to donate you a device to help make it first class!?

I will also point out for reference that I'm using bazzite on the device.

**Comment by streetpea** [comment]
@chino yes it has the capability to read touchscreen inputs, touchpad inputs for controllers, mouse inputs (holding down left click) and dpad inputs when dpad touch is enabled.
